src/ray/common/id.h

```diff
@@ -78,78 +139,165 @@ class UniqueID : public BaseID<UniqueID> {
 
 class JobID : public BaseID<JobID> {
  public:
-  static constexpr int64_t length = 4;
+  static constexpr int64_t LENGTH = 4;
 
-  // TODO(qwang): Use `uint32_t` to store the data.
   static JobID FromInt(uint32_t value);
 
-  static size_t Size() { return length; }
+  static size_t Size() { return LENGTH; }
 
   static JobID FromRandom() = delete;
 
   JobID() : BaseID() {}
 
  private:
-  uint8_t id_[length];
+  uint8_t id_[LENGTH];
+};
+
+class ActorID : public BaseID<ActorID> {
+ private:
+  static constexpr size_t UNIQUE_BYTES_LENGTH = 4;
+
+ public:
+  static constexpr size_t LENGTH = UNIQUE_BYTES_LENGTH + JobID::LENGTH;
+
+  static size_t Size() { return LENGTH; }
+
+  static ActorID FromRandom(const JobID &job_id);
+
+  static ActorID FromRandom() = delete;
+
+  ActorID() : BaseID() {}
+
+  JobID JobId() const;
+
+ private:
+  uint8_t id_[LENGTH];
 };
 
 class TaskID : public BaseID<TaskID> {
+ private:
+  static constexpr size_t UNIQUE_BYTES_LENGTH = 6;
+
  public:
+  static constexpr size_t LENGTH = UNIQUE_BYTES_LENGTH + ActorID::LENGTH;
+
   TaskID() : BaseID() {}
-  static size_t Size() { return kTaskIDSize; }
+
+  static size_t Size() { return LENGTH; }
+
   static TaskID ComputeDriverTaskId(const WorkerID &driver_id);
 
+  static TaskID FromRandom(const ActorID &actor_id);
+
  private:
-  uint8_t id_[kTaskIDSize];
+  uint8_t id_[LENGTH];
 };
 
+// TODO(qwang): Add complete designing to describe structure of ID.
 class ObjectID : public BaseID<ObjectID> {
+private:
+  static constexpr size_t INDEX_BYTES_LENGTH = sizeof(ObjectIDIndexType);
+
+  static constexpr size_t FLAGS_BYTES_LENGTH = sizeof(ObjectIDFlagsType);
+
  public:
+  /// The maximum number of objects that can be returned by a task when finishing
+  /// execution. An ObjectID's bytes are split into the task ID itself and the
+  /// index of the object's creation.
+  static constexpr int64_t MAX_OBJECT_INDEX = ((int64_t) 1 << kObjectIdIndexSize) - 1;
+
+  static constexpr size_t LENGTH = INDEX_BYTES_LENGTH + FLAGS_BYTES_LENGTH + TaskID::LENGTH;
+
   ObjectID() : BaseID() {}
-  static size_t Size() { return kUniqueIDSize; }
+
+  /// The maximum index of object. It also means the number of objects(put or return)
+  /// of one task.
+  static uint64_t MaxObjectIndex() { return MAX_OBJECT_INDEX; }
```

Comment:
Is this method needed only because Cython cannot access `MAX_OBJECT_INDEX`?

Reply:
Yes.